src/AzSK.ADO/Framework/Core/STMapping/ServiceMappingCacheHelper.ps1

```diff
@@ -95,7 +95,7 @@ class ServiceMappingCacheHelper {
         }
         if($resourceType -eq "VariableGroup" -or $resourceType -eq "SecureFile" -and [string]::IsNullOrEmpty($resourceId)) 
         {
-            return 'OrgName eq ''{0}'' and PipelineType eq ''{1}''' -f $this.OrganizationName, $pipelineType;
+            return 'OrgName eq ''{0}'' and ProjectID eq ''{1}'' and ResourceType eq ''{2}'' and ServiceTreeID ne ''UNMAPPED''' -f $this.OrganizationName, $projectID, $resourceType;
```

Comment:
As discussed, Please retain the last mapped service tree id of any unmapped resource.